### PR TITLE
Revert "Revert "[Doc] [Jobs] Document `JobDetails`, `JobType` and `DriverInfo`""

### PR DESCRIPTION
**Pull request by architkulkarni**
Reverts ray-project/ray#30499.  The reason this PR was originally reverted because it caused Jobs runtime env tests to fail, because runtime_env CI tests were installing `"ray"` into the runtime_env conda environment instead of `"ray[default]"`, which is required for Jobs.  This PR fixes this by installing `"ray[default]"` in runtime_env CI tests.

Original PR description: 

The classes `JobDetails`, `JobType` and `DriverInfo` are exposed to the user because they are the return type of `JobSubmissionClient.get_job_info`, so we need to document them. 

This PR adds them to the API reference.

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]
stamp

**Review by sihanwang41 — APPROVED** [review]

File: dashboard/utils.py [hunk 1]
```diff
@@ -123,12 +123,12 @@ def get_all_modules(module_type):
             logger.info(
                 f"Module {name} cannot be loaded because "
                 "we cannot import all dependencies. Download "
```
Comment (dmatrix) [inline dashboard/utils.py:125]: Can we say. "Install this module using `pip....` instead of download?

**Review by dmatrix — APPROVED** [review]
Made a minor or nit suggestion. Otherwise, LGTM

File: dashboard/utils.py (hunk 1 above)
Comment (architkulkarni) [reply dashboard/utils.py]: Done!

**Comment by architkulkarni** [comment]
`doc:plot_pong_example` unrelated and was flaky on master
`tests:test_cli` unrelated and was broken on master

**Pull request merged**